### PR TITLE
Add semantic headings to pages with multiple cards

### PR DESCRIPTION
**Pull request by mylesw27**
This PR will address one of the tasks from issue #10446 by adding semantic headings to pages with multiple cards. In a previous PR, we added `<h1>` tags to all pages. However, on pages with multiple cards, these `<h1>` headings were added as `.sr-only`. This means that the current headings for tables are not using semantic headings. This PR intends to add `<h2>` tags where applicable and make navigating these pages easier for anyone using a screen reader. 

Please note, this PR does not add headings to the questions page. This page has several cards/panels and is being handled in a separate PR. Please let me know if you are able to find any other pages I may have missed.

### CODE REVIEW

File: apps/prairielearn/src/pages/home/home.html.ts [hunk 1]
```diff
@@ -55,7 +55,6 @@ export function Home({
         </header>
 
         <main id="content" class="flex-grow-1">
-          <h1 class="sr-only">PrairieLearn Homepage</h1>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/home/home.html.ts:58]: Why was this removed?

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:1]: I can't comment on the line, but: the `<h1>` here doesn't handle group work correctly.

File: apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts [hunk 2]
```diff
@@ -94,7 +94,9 @@ export function InstructorAssessmentQuestionStatistics({
 
           <div class="card mb-4">
             <div class="card-header bg-primary text-white d-flex align-items-center">
-              ${resLocals.assessment_set.name} ${resLocals.assessment.number}: Question difficulty
+              <h2>
+                ${resLocals.assessment_set.name} ${resLocals.assessment.number}: Question difficulty
+              </h2>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts:99]: This could possibly be improved when we tackle https://github.com/PrairieLearn/PrairieLearn/issues/10447. I don't think it's necessary to repeat the name/number for every card on the page. Not requesting a change here; let's discuss separately.

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts [hunk 3]
```diff
@@ -179,25 +179,27 @@ export function InstructorCourseAdminSharing({
...
-                            <i class="fas fa-share-nodes" aria-hidden="true"></i>
-                            <span class="d-none d-sm-inline">Choose Sharing Name</span>
-                          </button>
-                          ${ChooseSharingNameModal({
-                            canChooseSharingName,
-                            csrfToken: resLocals.__csrf_token,
-                          })}
-                        `
-                      : ''}
+                    ${
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts:182]: Hmm, why did this change? Ditto for the below change in this file.

File: apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export function InstructorIssues({
             <div class="card-header bg-primary text-white">
               <div class="d-flex flex-row align-items-center mb-2">
                 <div class="d-flex flex-column">
-                  <h1 class="h6 font-weight-normal mb-0">Issues</h1>
+                  <h1>Issues</h1>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts:103]: Gonna guess you didn't test this:

Screenshot 2024-08-16 at 10 43 41

See my note here: https://github.com/PrairieLearn/PrairieLearn/pull/10485#discussion_r1719057190

File: apps/prairielearn/src/pages/home/home.html.ts (hunk 1 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/home/home.html.ts]: Removed inadvertently when testing screen reader functionality. Restored in [c5187bb](https://github.com/PrairieLearn/PrairieLearn/pull/10483/commits/c5187bb79e07a5a99f75ef5cfa5a9961d6781e2d).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:1]: Added conditional for reading group names in [c5187bb](https://github.com/PrairieLearn/PrairieLearn/pull/10483/commits/c5187bb79e07a5a99f75ef5cfa5a9961d6781e2d).

File: apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts (hunk 2 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts:99]: That is a great point. Let me know when you want to discuss these headings.

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts (hunk 3 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts]: Had an unclosed `<h2>` that was causing formatting issues. Corrected in [c5187bb](https://github.com/PrairieLearn/PrairieLearn/pull/10483/commits/c5187bb79e07a5a99f75ef5cfa5a9961d6781e2d).

File: apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts (hunk 4 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts]: You would be correct. Added the classes back (and tested this time) in [c5187bb](https://github.com/PrairieLearn/PrairieLearn/pull/10483/commits/c5187bb79e07a5a99f75ef5cfa5a9961d6781e2d).

File: apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts:99]: Maybe add this as a note to that issue? I think we can discuss some of this in a future dev meeting.

Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts:99]: Added a note to that issue.

**Review by nwalters512 — APPROVED** [review]
